ls1intum/Artemis
# `Programming exercises`: Allow automatically graded exercises where all test cases give 0 points

# This PR is based on #5994

### Checklist
#### General
#### Server
#### Client

### Motivation and Context
#6093

### Description
This PR removes the weight check that depends on the assessment type. Now, only negative weights lead to an error.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Programming Exercise

1. Navigate to the grading page of the exercise and set all tests to a weight of 0.
2. Check that saving works.
3. Check the solution:
a) It should show 100% if all tests are successful
b) It shows less then 100% if some test are not successful or if there are SCA deductions
5. Participate as a Student. You should always have 0 points, no matter how many tests passed.

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
